lib/hotel_block.rb

```diff
@@ -0,0 +1,96 @@
+require 'date'
+# # require_relative 'hotel.rb'
+
+module Booking
+  class Block
+
+    attr_reader :checkin_date, :checkout_date, :requested_rooms, :discounted_rate, :party_name, :room_block, :block_of_rooms, :all_rooms, :max_rooms, :room_numbers, :hotel, :all_reservations
+    
+    def initialize(checkin_date, checkout_date, requested_rooms, discounted_rate, hotel: nil)
+    @checkin_date = checkin_date
+    @checkout_date = checkout_date 
+    @block_of_rooms ||= []
```

Comment:
The `||=` syntax is saying something along the lines of "if `@block_of_rooms` is `nil`, assign it to the thing on the right, which is `[]`". Since we can rely on `@block_of_rooms` being `nil` here (since it's the `initialize` method), it's probably okay to let this be regular assignment `=`